src/communication/dryrun_global_policy.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+#include <vector>
+
+#include <communication/gathered_vector.hpp>
+#include <util/span.hpp>
+#include <spike.hpp>
+
+namespace nest {
+namespace mc {
+namespace communication {
+
+extern int dryrun_num_local_cells;
+extern int dryrun_communicator_size;
+
+struct dryrun_global_policy {
+    template <typename Spike>
+    static gathered_vector<Spike>
+    gather_spikes(const std::vector<Spike>& local_spikes) {
+        using util::make_span;
+        using count_type = typename gathered_vector<Spike>::count_type;
+
+        // Build the global spike list by replicating the local spikes for each
+        // "dummy" domain.
+        auto num_spikes_local  = local_spikes.size();
+        auto num_spikes_global = size()*num_spikes_local;
+        std::vector<Spike> global_spikes(num_spikes_global);
+        std::vector<count_type> partition(size()+1);
+
+        for (auto rank: make_span(0u, size())) {
+            auto first_cell = rank*dryrun_num_local_cells;
+            auto first_spike = rank*num_spikes_local;
+            for (auto i: make_span(0, num_spikes_local)) {
```

Comment:
Shouldn't this b const auto&& ? Unless you know it collapses to a primitive -- and even then, it should be general.